ts/features/payments/onboarding/components/WalletOnboardingPaymentMethodsList.tsx

```diff
@@ -63,25 +69,32 @@ const PaymentMethodItem = ({
 const WalletOnboardingPaymentMethodsList = ({
   paymentMethods,
   onSelectPaymentMethod,
-  isLoading,
+  isLoadingMethods,
+  isLoadingWebView,
   header
-}: OwnProps) => (
-  <FlatList
-    removeClippedSubviews={false}
-    contentContainerStyle={IOStyles.horizontalContentPadding}
-    data={paymentMethods}
-    keyExtractor={item => item.name}
-    ListHeaderComponent={header}
-    ListFooterComponent={renderListFooter(isLoading)}
-    ItemSeparatorComponent={() => <Divider />}
-    renderItem={({ item }) => (
-      <PaymentMethodItem
-        paymentMethod={item}
-        onPress={() => onSelectPaymentMethod(item)}
-      />
-    )}
-  />
-);
+}: OwnProps) => {
+  const selectedPaymentMethodId = useIOSelector(
+    walletOnboardingSelectedPaymentMethodSelector
+  );
+  return (
+    <FlatList
+      removeClippedSubviews={false}
+      contentContainerStyle={IOStyles.horizontalContentPadding}
+      data={paymentMethods}
+      keyExtractor={item => item.id}
+      ListHeaderComponent={header}
+      ListFooterComponent={renderListFooter(isLoadingMethods)}
+      ItemSeparatorComponent={() => <Divider />}
+      renderItem={({ item }) => (
+        <PaymentMethodItem
+          paymentMethod={item}
+          isLoading={isLoadingWebView && item.id === selectedPaymentMethodId}
+          onPress={() => onSelectPaymentMethod(item)}
+        />
+      )}
+    />
+  );
+};
```

Comment:
```suggestion
 const isMethodLoading = (itemId: string) =>
    isLoadingWebView && itemId === selectedPaymentMethodId;
  const ListFooter = () => renderListFooter(isLoadingMethods);
  return (
    <FlatList
      removeClippedSubviews={false}
      contentContainerStyle={IOStyles.horizontalContentPadding}
      data={paymentMethods}
      keyExtractor={item => item.id}
      ListHeaderComponent={header}
      ListFooterComponent={<ListFooter />}
      ItemSeparatorComponent={() => <Divider />}
      renderItem={({ item }) => (
        <PaymentMethodItem
          paymentMethod={item}
          isLoading={isMethodLoading(item.id)}
          onPress={() => onSelectPaymentMethod(item)}
        />
      )}
    />
  );
};
```
I think moving logic checks outside of the render makes it look a little neater, what do you think?
Also, are you sure you want to handle the loading through the footer instead of swapping the render function when `isLoadingMethods===true`?

Reply:
I think now is better with your suggestions: 4e0a1b973002d6751794fb8db482697d55b6dc6b